src/FormControl.js

```diff
@@ -0,0 +1,70 @@
+import classNames from 'classnames';
+import React from 'react';
+import elementType from 'react-prop-types/lib/elementType';
+
+import bootstrapUtils, { bsClass } from './utils/bootstrapUtils';
+
+import FormControlFeedback from './FormControlFeedback';
+import FormControlStatic from './FormControlStatic';
+
+const propTypes = {
+  componentClass: elementType,
+  /**
+   * Only relevant if `componentClass` is `'input'`.
+   */
+  type: React.PropTypes.string,
+  /**
+   * Uses `controlId` from `<FormGroup>` if not explicitly specified.
+   */
+  id: React.PropTypes.string,
+};
+
+const defaultProps = {
+  componentClass: 'input',
+};
+
+const contextTypes = {
+  $bs_formGroup: React.PropTypes.object,
+};
+
+class FormControl extends React.Component {
+  getControlId(formGroup) {
+    return formGroup && formGroup.controlId;
+  }
+
+  render() {
+    const {
+      componentClass: Component,
+      type,
+      id = this.getControlId(this.context.$bs_formGroup),
```

Comment:
if `controlID` exists does it makes more sense to use it over the specified `props.id` so that there isn't a potential label with an `htmlFor` pointing to nothing by mistake?

Or maybe just warn?